include/pybind11/cast.h

```diff
@@ -735,12 +822,14 @@ template <typename type, typename holder_type> class type_caster_holder : public
         }
 
         if (convert) {
-            for (auto &converter : typeinfo->implicit_conversions) {
+            // Walk through the implicit python conversions looking for one that works.
+            for (auto &converter : typeinfo->implicit_python_conversions) {
                 temp = object(converter(src.ptr(), typeinfo->type), false);
```

Comment:
This is super minor: I would prefer it if the fields were called ``implicit_conversions_python`` and ``implicit_conversions_cpp``